code/go/0chain.net/smartcontract/dbs/event/search.go

```diff
@@ -0,0 +1,40 @@
+package event
+
+import (
+	"database/sql"
+	"strconv"
+)
+
+// GetGenericSearchType returns kind of search query
+func (edb *EventDb) GetGenericSearchType(query string) (string, error) {
+	var queryType string
+	var round int
+
+	round, err := strconv.Atoi(query)
+	if err != nil {
```

Comment:
if it errors, we should not even query in blocks table

Reply:
Done